framework/wazuh/rbac/orm.py

```diff
@@ -2875,206 +3218,21 @@ def migrate_data(self, source: str, target: str, from_id: int = None, to_id: int
-                            role_policy_manager.add_policy_to_role(role_id=role_policy.role_id,
-                                                                   policy_id=new_policy_id,
-                                                                   position=role_policy.level,
-                                                                   created_at=role_policy.created_at,
-                                                                   force_admin=True)
-                        logger.info(f"All relationships were migrated to the new policy {new_policy_id}")
-
-        old_user_roles = self.get_data(source, UserRoles, UserRoles.user_id, UserRoles.role_id, from_id=from_id,
-                                       to_id=to_id)
-        old_user_roles = sorted(old_user_roles, key=lambda item: item.level)
-        with UserRolesManager(self.sessions[target]) as user_role_manager:
-            for user_role in old_user_roles:
-                user_id = user_role.user_id
-                role_id = user_role.role_id
-                # Look for the ID of a default resource from the old database in the new database using its name
-                # This allows us to keep the relationship if the related default resource now has a different id
-                if int(user_id) <= MAX_ID_RESERVED:
-                    try:
-                        user_name = self.get_table(self.sessions[source], User).filter(
-                            User.id == user_id).first().username
-                        user_id = AuthenticationManager(self.sessions[target]).get_user(username=user_name)['id']
-                    except TypeError:
-                        logger.warning(f"User {user_id} ({user_name}) no longer exists. Removing affected "
-                                       "user-role relationships")
-                        continue
-
-                if int(role_id) <= MAX_ID_RESERVED:
-                    try:
-                        role_name = self.get_table(self.sessions[source], Roles).filter(
-                            Roles.id == role_id).first().name
-                        role_id = RolesManager(self.sessions[target]).get_role(name=role_name)['id']
-                    except TypeError:
-                        logger.warning(f"Role {role_id} ({role_name}) no longer exists. Removing affected "
-                                       "user-role relationships")
-                        continue
-
-                user_role_manager.add_role_to_user(user_id=user_id,
-                                                   role_id=role_id,
-                                                   created_at=user_role.created_at,
-                                                   force_admin=True)
+        for manager in list_if_must_run_resources:
+            with manager(self.sessions[target]) as resource_manager:
+                resource_manager.migrate_data(self, source, target, from_id=from_id, to_id=to_id)
```

Comment:
```suggestion
        with AuthenticationManager(self.sessions[target]) as auth_manager:
            auth_manager.migrate_data(self, source, target, from_id=from_id, to_id=to_id)
```

Reply:
Changed in 5afee8f5d4b7c3771a18a307f32df1ffe64af313.